resources/js/components/catalog-new/Filter.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+    <div class="tw-relative">
+        <slot />
+    </div>
+</template>
+
+<script>
+import NewCustomSelect from './NewCustomSelect.vue'
+import NewMobileCustomSelect from './NewMobileCustomSelect.vue'
+
+export default {
+    props: {
+        authors: Array,
+    },
+    components: { NewCustomSelect, NewMobileCustomSelect },
+    data() {
+        return {
+            openedFilter: null,
+            isExtendedOpen: true,
+        }
+    },
+    computed: {
+        filters() {
+            return {
+                authors: {
+                    list: this.authors.map((author) => ({
+                        ...author,
+                        checked: this.isSelectedMultiSelect('authors', author.name),
+                    })),
+                },
+                someOtherFilter: { list: [] },
+            }
+        },
+    },
+    methods: {
+        isSelectedMultiSelect(filterName, name) {
+            const urlQuery = this.$route.query
+            return urlQuery[filterName] && urlQuery[filterName].includes(name)
+        },
+        toggleIsExtendedOpen() {
+            this.isExtendedOpen = !this.isExtendedOpen
+        },
+        clearSelection(filterName) {
+            const { [filterName]: removedFilterName, ...queryWithoutFilterName } = this.$route.query  
+            this.$router.push({
+                path: 'katalog-new',
+                query: queryWithoutFilterName,
+            })
+            this.filters[filterName].search = null
+            this.openedFilter = null
+        },
+        setOpenedFilter(name) {
+            this.openedFilter = name
+        },
+        handleChangeMultiSelect(filterName, value, selected) {
+            const urlQuery = this.$route.query
+            const filterNameVals = urlQuery[filterName] || []
+
+            this.$router.push({
+                path: 'katalog-new',
+                query: {
+                    ...urlQuery,
+                    [filterName]: selected
+                        ? [...filterNameVals, value]
+                        : filterNameVals.filter((filterNameVal) => value !== filterNameVal),
+                },
+            })
+        },
```

Comment:
Removing keys from objects is a little cumbersome in JS. I think the whole process will be easier to reason about when you just send an array of selections in each event, i.e.

```suggestion
        handleMultiSelectChange(filterName, selectedValues) {
            this.$router.push({
                query: {
                    ...this.$route.query,
                    [filterName]: selectedValues,
                },
            })
        },
```

Reply:
OK, I'll take a look later on